app/assets/javascripts/controllers/ansible_repository/repository_form_controller.js

```diff
@@ -0,0 +1,119 @@
+ManageIQ.angular.app.controller('repositoryFormController', ['$http', '$scope', 'repositoryId', 'miqService', 'API', function($http, $scope, repositoryId,  miqService, API) {
+  var vm = this;
+
+  var init = function() {
+    vm.afterGet = false;
+
+    vm.repositoryModel = {
+      name: '',
+      description: '',
+      scm_type: 'git',
+      manager_resource: {},
+      scm_url: '',
+      scm_credentials: null,
+      scm_branch: '',
+      scm_clean: false,
+      scm_delete_on_update: false,
+      scm_update_on_launch: false,
+    };
+
+    API.get('/api/providers?collection_class=ManageIQ::Providers::EmbeddedAutomationManager')
+      .then(getManagerResource)
+      .catch(miqService.handleFailure);
+
+    vm.model = 'repositoryModel';
+
+    ManageIQ.angular.scope = vm;
+
+    $scope.newRecord = repositoryId === 'new';
+
+    vm.scm_credentials = [{name: __('Select credentials'), value: null}];
+    API.get('/api/authentications?collection_class=ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ScmCredential&expand=resources')
+      .then(getCredentials)
+      .catch(miqService.handleFailure);
+
+    if (repositoryId !== 'new') {
+      API.get('/api/configuration_script_sources/' + repositoryId)
+        .then(getRepositoryFormData)
+        .catch(miqService.handleFailure);
+    } else {
+      vm.afterGet = true;
+      vm.modelCopy = angular.copy( vm.repositoryModel );
+      miqService.sparkleOff();
+    }
+  };
+
+  $scope.cancelClicked = function() {
+    var message = repositoryId === 'new' ? __("Addition of Repository canceled by user.") : sprintf(__("Addition of Repository \"%s\" canceled by user."), vm.repositoryModel.name)
```

Comment:
this should be "Add of Repository cancelled by user." : "Edit of Repository \"%s\" cancelled by user."